source/slang/ir.cpp

```diff
@@ -3409,10 +3464,29 @@ namespace Slang
         removeFromParent();
         removeArguments();
 
+        // If this is a parent instruction then we had
+        // better remove all its children as well.
+        //
+        if(auto parentInst = as<IRParentInst>(this))
+        {
+            parentInst->removeAndDeallocateAllChildren();
+        }
+
         // Run destructor to be sure...
         this->~IRInst();
     }
 
+    void IRParentInst::removeAndDeallocateAllChildren()
+    {
+        IRInst* nextChild = nullptr;
+        for( IRInst* child = getFirstChild(); child; child = nextChild )
```

Comment:
I suppose Deallocate is appropriate here in so far in principal could deallocate even though in our current impl we actually don't.

Reply:
I'm just following the naming of the existing `removeAndDeallocate()` function, so if we want to rename one we should do the other also. :)